### ISSUE TITLE
[Bug] Screen resolution and fingerprint not saved in Server tracking mode

### ISSUE THREAD

**Issue by parhumm** [slimstat/wp-slimstat#163]
## Bug: Screen resolution and fingerprint not saved in Server tracking mode

**Reported via**: [wp.org support thread](https://wordpress.org/support/topic/missing-info-on-the-real-time-page/)

### Summary

In **Server** tracking mode, screen resolution, fingerprint, server latency, page performance, and timezone offset data collected by the JavaScript tracker are never saved to the database. The data is silently discarded due to a state management regression in `Ajax::handle()`.

### Root Cause

**Fault Location**: `src/Tracker/Ajax.php:174`

In the Server mode path of `Ajax::handle()`, the local `$stat` variable accumulates the record `id` (line 100), processed IP (line 109), user data (lines 112-124), and resource URL (lines 130-170). However, before this data is persisted:

1. `Session::ensureVisitId(true)` is called at line 173, which internally calls `get_stat()` + `set_stat()` — overwriting the class property with `visit_id` but without the locally accumulated data
2. Line 174: `$stat = \wp_slimstat::get_stat()` **re-reads from the class property**, discarding the local `$stat` — including the critical `id` field
3. `Utils::getClientInfo()` at line 181 adds screen resolution and fingerprint to the now-incomplete `$stat`
4. `Storage::updateRow($stat)` at line 257 receives `$stat` without `id` → returns `false` silently (Storage.php:27-28)

**Timeline**: This regression was introduced in commit `d6b8bb9c` (Nov 5, 2025) which added line 174 to pick up `visit_id` from `ensureVisitId()`, inadvertently discarding all other locally accumulated state.

**Note**: The original code (commit `154f6ff3`, Sep 9, 2025) used direct property access (`\wp_slimstat::$stat['id']`), so the `id` was always part of the class property. The getter/setter refactor (`c9c5e322`) introduced local variables, and the subsequent GDPR compliance commit (`d6b8bb9c`) broke the flow by re-reading from the class property.

### Causal Chain

```
Trigger: Visitor loads page with Server tracking mode enabled
  → Entry Point: Ajax::handle() receives JS callback with stat id
    → set_stat() called at line 92 (before id is set at line 100)
      → ensureVisitId() at line 173 reads/writes class property (no id)
        → get_stat() at line 174 overwrites local $stat (loses id)
          → updateRow() at line 257 fails silently (no id to update)
            → Impact: resolution, fingerprint, latency, performance, tz_offset never saved
```

### Reproduction

**Preconditions**: WordPress + SlimStat, Tracking Mode = "Server"

1. Visit any tracked page on the website
2. Server-side tracking creates a DB record with an `id`
3. JS tracker loads and sends `sw`, `sh`, `bw`, `bh`, `fh` (screen/fingerprint data) via AJAX
4. Ajax handler processes the request but `updateRow()` fails because `$stat['id']` is empty
5. Navigate to SlimStat → Real Time — screen resolution and fingerprint columns are empty

**Expected**: Screen resolution and fingerprint appear in Real Time view
**Actual**: Fields are empty for all Server-mode pageviews

### Affected Data

All client-side metrics in Server mode:
- `screen_width` / `screen_height` / `resolution`
- `fingerprint`
- `server_latency`
- `page_performance`
- `tz_offset`

### Suggested Fix

Replace line 174 with a targeted merge that preserves the locally accumulated `$stat`:

```php
// After ensureVisitId() — merge only visit_id from class property
$classStat = \wp_slimstat::get_stat();
$stat['visit_id'] = $classStat['visit_id'] ?? 0;
```

Or alternatively, call `\wp_slimstat::set_stat($stat)` immediately after line 125 (after `id`, IP, and user data are set) so the class property includes all accumulated data before `ensureVisitId()` reads it.

### Code References

| File | Line(s) | Description |
|------|---------|-------------|
| `src/Tracker/Ajax.php` | 92 | `set_stat($stat)` called before `id` is set |
| `src/Tracker/Ajax.php` | 100 | `$stat['id']` set on local variable only |
| `src/Tracker/Ajax.php` | 174 | `$stat = get_stat()` overwrites local — **regression** |
| `src/Tracker/Ajax.php` | 257 | `Storage::updateRow($stat)` fails silently |
| `src/Tracker/Storage.php` | 27-28 | Returns `false` when `$stat['id']` is empty |
| `src/Tracker/Utils.php` | 316-360 | `getClientInfo()` extracts resolution + fingerprint |

### Severity

**High** — Major data loss affecting all Server-mode users. No workaround exists except switching to Client mode (which the reporter notes misses many visitors).
